library/src/main/java/com/squareup/timessquare/CalendarPickerView.java

```diff
@@ -794,45 +804,45 @@ private MonthCellWithMonthIndex getMonthCellWithIndexByDate(Date date) {
     return null;
   }
 
-  private class MonthAdapter extends BaseAdapter {
+  private class MonthAdapter extends RecyclerView.Adapter<MonthAdapter.MonthViewHolder> {
     private final LayoutInflater inflater;
 
     private MonthAdapter() {
       inflater = LayoutInflater.from(getContext());
     }
 
-    @Override public boolean isEnabled(int position) {
-      // Disable selectability: each cell will handle that itself.
-      return false;
-    }
-
-    @Override public int getCount() {
-      return months.size();
+    @Override
+    public MonthViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+      return new MonthViewHolder(MonthView.create(parent, inflater, weekdayNameFormat, listener, today, dividerColor,
+              dayBackgroundResId, dayTextColorResId, titleTextColor, displayHeader,
+              headerTextColor, decorators, locale, dayViewAdapter));
     }
 
-    @Override public Object getItem(int position) {
-      return months.get(position);
+    @Override
+    public void onBindViewHolder(MonthViewHolder holder, int position) {
+      holder.monthView.setTag(R.id.day_view_adapter_class, dayViewAdapter.getClass());
+      holder.monthView.init(months.get(position), cells.get(position), displayOnly, titleTypeface,
+              dateTypeface);
     }
 
     @Override public long getItemId(int position) {
       return position;
     }
 
-    @Override public View getView(int position, View convertView, ViewGroup parent) {
-      MonthView monthView = (MonthView) convertView;
-      if (monthView == null
-             || !monthView.getTag(R.id.day_view_adapter_class).equals(dayViewAdapter.getClass())) {
-        monthView =
-            MonthView.create(parent, inflater, weekdayNameFormat, listener, today, dividerColor,
-                dayBackgroundResId, dayTextColorResId, titleTextColor, displayHeader,
-                headerTextColor, decorators, locale, dayViewAdapter);
-        monthView.setTag(R.id.day_view_adapter_class, dayViewAdapter.getClass());
-      } else {
-        monthView.setDecorators(decorators);
+    @Override
+    public int getItemCount() {
+      return months.size();
+    }
+
+    class MonthViewHolder extends RecyclerView.ViewHolder  {
+
```

Comment:
super-nit: remove blank line.  and remove the extra space in the previous line (before the `{`)